src/lib/y2storage/proposal/autoinst_devices_planner.rb

```diff
@@ -0,0 +1,155 @@
+        end
+
+        if result.empty?
+          raise Error, "No partitions specified"
+        end
+
+        checker = BootRequirementsChecker.new(devicegraph, planned_devices: result)
+        result.concat(checker.needed_partitions)
+
+        result
+      end
+
+    protected
+
+      # @return [Devicegraph] Starting devicegraph
+      attr_reader :devicegraph
+
+      # Returns an array of planned partitions for a given disk
+      #
+      # @param disk         [Disk] Disk to place the partitions on
+      # @param partitioning [Hash] Partitioning specification from AutoYaST
+      # @return [Array<Planned::Partition>] List of planned partitions
+      def planned_for_disk(disk, spec)
+        result = []
+        spec.fetch("partitions", []).each do |part_description|
+          # TODO: fix Planned::Partition.initialize
+          part = Y2Storage::Planned::Partition.new(nil, nil)
+          part.disk = disk.name
+          # part.bootable no está en el perfil (¿existe lógica?)
+          part.filesystem_type = filesystem_for(part_description["filesystem"])
+          # TODO: set the correct id based on the filesystem type (move to Partition class?)
+          part.partition_id = 131
+          if part_description["crypt_fs"]
+            part.encryption_password = part_description["crypt_key"]
+          end
+          part.mount_point = part_description["mount"]
+          part.label = part_description["label"]
+          part.uuid = part_description["uuid"]
+          if part_description["create"] == false
+            partition_to_reuse = find_partition_to_reuse(devicegraph, part_description)
+            if partition_to_reuse
+              part.reuse = partition_to_reuse.name
+              part.reformat = !!part_description["format"]
+            end
+            # TODO: possible errors here
+            #   - missing information about what device to use
+            #   - the specified device was not found
+          end
+
+          # Sizes: leave out reducing fixed sizes and 'auto'
+          min_size, max_size = sizes_for(part_description, disk)
+          part.min_size = min_size
+          part.max_size = max_size
+          result << part
+        end
+
+        result
+      end
+
+      # Regular expression to detect which kind of size is being used in an
+      # AutoYaST <size> element
+      SIZE_REGEXP = /([\d,.]+)?([a-zA-Z%]+)/
+
+      # Returns min and max sizes for a partition specification
+      #
+      # @param description [Hash] Partition specification from AutoYaST
+      # @return [[DiskSize,DiskSize]] min are max sizes for the given partition
+      #
+      # @see SIZE_REGEXP
+      def sizes_for(part_spec, disk)
+        normalized_size = part_spec["size"].to_s.strip.downcase
+
+        if normalized_size == "max" || normalized_size.empty?
+          return [disk.min_grain, DiskSize.unlimited]
+        end
+
+        _all, number, unit = SIZE_REGEXP.match(normalized_size).to_a
```

Comment:
another option is 
`number, unit = SIZE_REGEXP.match(normalized_size).values_at(1,2)`